packages/@aws-cdk/aws-appsync/lib/graphqlapi-base.ts

```diff
@@ -0,0 +1,146 @@
+import { ITable } from '@aws-cdk/aws-dynamodb';
+import { IFunction } from '@aws-cdk/aws-lambda';
+import { CfnResource, IResource, Resource } from '@aws-cdk/core';
+import { DynamoDbDataSource, HttpDataSource, LambdaDataSource, NoneDataSource } from './data-source';
+
+/**
+ * Interface for GraphQL
+ */
+export interface IGraphQLApi extends IResource {
+
+  /**
+   * an unique AWS AppSync GraphQL API identifier
+   * i.e. 'lxz775lwdrgcndgz3nurvac7oa'
+   * @attribute
+   */
+  readonly apiId: string;
+
+  /**
+   * the ARN of the API
+   *
+   * @attribute
+   */
+  readonly arn: string;
+
+  /**
+   * add a new dummy data source to this API. Useful for pipeline resolvers
+   * and for backend changes that don't require a data source.
+   * @param name The name of the data source
+   * @param description The description of the data source
+   */
+  addNoneDataSource(name?: string, description?: string): NoneDataSource;
+
+  /**
+   * add a new DynamoDB data source to this API
+   * @param name The name of the data source
+   * @param description The description of the data source
+   * @param table The DynamoDB table backing this data source
+   */
+  addDynamoDbDataSource(name: string, description: string, table: ITable): DynamoDbDataSource;
```

Comment:
it sounds like name and description should be optional for any given data source? we should be able to create sensible defaults if they're required under the hood.

I'm of the opinion that the signature should require a table and options to set the rest... but I'll defer to whatever @MrArnoldPalmer thinks

Reply:
we talked offline and cleared up the confusion. I thought the implementation would end up looking like this:

```ts
api.addDynamoDbDataSource('tableDs', {
  table: table,
  description: 'my table',
});
```

whereas @shivlaks was suggesting this:

```ts
api.addDynamoDbDataSource(table, {
  name: 'tableDs',
  description: 'my table',
});
```

where the `options` is optional. I agree, this is MUCH cleaner